halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/RequestGenerateService.java

```diff
@@ -124,6 +118,22 @@ protected void writeServiceSettings(
     }
   }
 
+  private void writeFile(String deploymentName, String filePath, MultipartFile fileContents) {
+    Path target = Paths.get(baseDirectory.toString(), deploymentName, filePath).normalize();
+    if (!target.startsWith(baseDirectory.toString())) {
+      return;
```

Comment:
If the user is attempting through `..` to write outside the config directory, we should throw an exception.